sdks/python/apache_beam/examples/inference/multi_language/expansion_service/run_inference_expansion.py

```diff
@@ -0,0 +1,209 @@
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# pytype: skip-file
+
+import argparse
+import logging
+import signal
+import sys
+import typing
+
+import grpc
+
+import apache_beam as beam
+from apache_beam.coders import RowCoder
+from apache_beam.ml.inference.base import KeyedModelHandler
+from apache_beam.ml.inference.base import PredictionResult
+from apache_beam.ml.inference.base import RunInference
+from apache_beam.ml.inference.pytorch_inference import PytorchModelHandlerKeyedTensor
+from apache_beam.pipeline import PipelineOptions
+from apache_beam.portability.api import beam_artifact_api_pb2_grpc
+from apache_beam.portability.api import beam_expansion_api_pb2_grpc
+from apache_beam.portability.api import external_transforms_pb2
+from apache_beam.runners.portability import artifact_service
+from apache_beam.runners.portability import expansion_service
+from apache_beam.transforms import fully_qualified_named_transform
+from apache_beam.transforms import ptransform
+from apache_beam.transforms.environments import PyPIArtifactRegistry
+from apache_beam.transforms.external import ImplicitSchemaPayloadBuilder
+from apache_beam.utils import thread_pool_executor
+from transformers import BertConfig
+from transformers import BertForMaskedLM
+from transformers import BertTokenizer
+
+# This script provides an expansion service for a run inference transform
+# with pre and post processing.
+# The model used is a BertLM, base uncased model.
+_LOGGER = logging.getLogger(__name__)
+
+# A transform that runs inference on a Bertmodel.
+TEST_RUN_BERT_URN = "beam:transforms:xlang:test:run_bert"
+
+
+@ptransform.PTransform.register_urn(TEST_RUN_BERT_URN, None)
+class RunInferenceTransform(ptransform.PTransform):
+  class PytorchModelHandlerKeyedTensorWrapper(PytorchModelHandlerKeyedTensor):
+    """Wrapper to PytorchModelHandler to limit batch size to 1.
+        The tokenized strings generated from BertTokenizer may have different
+        lengths, which doesn't work with torch.stack() in current RunInference
+        implementation since stack() requires tensors to be the same size.
+        Restricting max_batch_size to 1 means there is only 1 example per
+        `batch` in the run_inference() call.
+        """
+    def batch_elements_kwargs(self):
+      return {'max_batch_size': 1}
```

Comment:
Does Bert normally handle batches of elements? Would it make sense to use a custom inference function instead of a full wrapper? A custom inference function allows you to supply a function to run instead of the default function (https://github.com/apache/beam/blob/f16d5d51b7551c12a854833676fc5a20bf8b6702/sdks/python/apache_beam/ml/inference/pytorch_inference.py#L145)

Reply:
I just took over the same logic from this notebook.

https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/inference/pytorch_language_modeling.py#L153